src/App.jsx

```diff
@@ -14,33 +17,102 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button type="button" className="button is-info is-light">
-        Sort alphabetically
-      </button>
+const getPreparedGoods = (goods, { sortedField, reversed }) => {
+  const preparedGoods = [...goods];
 
-      <button type="button" className="button is-success is-light">
-        Sort by length
-      </button>
+  if (sortedField) {
+    preparedGoods.sort((good1, good2) => {
+      switch (sortedField) {
+        case 'alphabet':
+          return good1.localeCompare(good2);
+        case 'length':
+          return good1.length - good2.length;
+        default:
+          return 0;
+      }
+    });
+  }
 
-      <button type="button" className="button is-warning is-light">
-        Reverse
-      </button>
+  if (reversed) {
+    preparedGoods.reverse();
+  }
 
-      <button type="button" className="button is-danger is-light">
-        Reset
-      </button>
-    </div>
+  return preparedGoods;
+};
+
+export const App = () => {
+  const [sortedField, setSortedField] = useState('');
+  const [reversed, setReversed] = useState('');
+  const visibleGoods = getPreparedGoods(goodsFromServer, {
+    sortedField,
+    reversed,
+  });
+
+  const isResetVisible = sortedField !== '' || reversed;
+
+  return (
+    <div className="section content">
+      <div className="buttons">
+        <button
+          onClick={() => setSortedField('alphabet')}
+          type="button"
+          className={cn(
+            {
+              'is-light': sortedField !== 'alphabet',
+            },
+            'button is-info',
+          )}
+        >
+          Sort alphabetically
+        </button>
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+        <button
+          onClick={() => setSortedField('length')}
+          type="button"
+          className={cn(
+            {
+              'is-light': sortedField !== 'length',
+            },
+            'button is-success',
+          )}
+        >
+          Sort by length
+        </button>
+
+        <button
+          onClick={() => setReversed(!reversed)}
```

Comment:
Updating state based on the previous state better to do via updater function:
https://react.dev/reference/react/useState#updating-state-based-on-the-previous-state